modules/pubmaticBidAdapter.js

```diff
@@ -219,26 +220,31 @@ function _parseAdSlot(bid) {
   }
   // check if size is mentioned in sizes array. in that case do not check for @ in adslot
   splits = slot.split('@');
-  if (splits.length != 2) {
-    if (!(sizesArrayExists)) {
+  /* if (splits.length != 2) {
+    if (!sizesArrayExists) {
       utils.logWarn('AdSlot Error: adSlot not in required format');
       return;
     }
-  }
+  } */
   bid.params.adUnit = splits[0];
-  if (splits.length > 1) { // i.e size is specified in adslot, so consider that and ignore sizes array
+  if (splits.length > 1) {
+    // i.e size is specified in adslot, so consider that and ignore sizes array
     splits = splits[1].split('x');
     if (splits.length != 2) {
       utils.logWarn('AdSlot Error: adSlot not in required format');
       return;
     }
     bid.params.width = parseInt(splits[0]);
     bid.params.height = parseInt(splits[1]);
-    delete bid.sizes;
-  } else if (sizesArrayExists) {
+    // delete bid.sizes;
+  } else {
+    bid.params.width = parseInt(bid.mediaTypes.banner.sizes[0][0]);
```

Comment:
In case of prebid, for a single banner format ad, if mediaTypes is not set by publisher, prebid adds that itself. So it should not happen that mediaTypes is not present. I can add a check, just in case, but what should be the behaviour in that scenario?